LEDGAME.ino

```diff
@@ -20,181 +23,187 @@ void setup()
+  return randomtime;
 }
 
+ //turn on random led
+int randomLedOn(int num)
+{
+   digitalWrite(ledInit[num], HIGH);   
+}
 
 
-//выключаю все кнопки 
-void led_off()
+//turn off all leds
+void ledOff(int num)
   {
-    // here you also can use a cycle
-   digitalWrite(ledInit[1], LOW);
-   digitalWrite(ledInit[2], LOW);
-   digitalWrite(ledInit[3], LOW);
-   digitalWrite(ledInit[0], LOW);
+    digitalWrite(ledInit[num], LOW);
   }  
 
-//выдаем результат в зависимости от нажатой кнопки 
-int buttWait(int num)
+//буфер с нажатыми кнопками
+int WhichButtPressed()
 {
-      buttPress(num);
-      if (num == butt[num])
-      {
-        Serial.print("                                                                  CORRECT! \n");
-        score += 1;
-      }    
-        else if (noButtPress == 1)
-        {
-          Serial.print("                                                          NO BUTTON PRESS DETECTED \n");
-        }
+  
+  for (int c = 1; c < 4; c++)
+  {
+   if (digitalRead(buttInit[c]) == HIGH)
+   {
+     butt[c] = c;
+     i = 1;
+   }
+  } 
+  
+}
+
 
-        return 0;
+int Logic(int num)
+{
+ if (butt[num] == num && i == 1)
+ {
+  score += 1;
+  Serial.print("                                                                  CORRECT! \n");
+  //correct
+ }
+  else if (butt[num] != num && i == 1)
+  {
+    Serial.print("                                                                 INCORRECT( \n");
+    //incorrect
+  }
+   else if (ledTimer.isTimerRunning() == false && i == 0)
+   {
+    Serial.print("                                                          NO BUTTON PRESS DETECTED \n");
+   }
 }
 
-//очистка буфера
-    void clearbutt()
+//clearing butt bufer
+    void clearButt()
     {
         butt[0] = 1; 
-      for (int i = 1; i < 5; i++)
+      for (int i = 1; i < 4; i++)
       {
         butt[i] = 0;
       }
     }
 
-//ждем кнопку старт
-void buttsWait()
+ //waiting for start button
+void startButtonWait()
 {
-  while (butt[0] == 0) 
-{
-  if(digitalRead(buttInit[0]) == HIGH){ 
-   butt[0] = 1; 
-}
+   while (butt[0] == 0) 
+   {
+     if(digitalRead(buttInit[0]) == HIGH)
+     { 
+      butt[0] = 1; 
+     }
+   }
 }
+
+void PrintScore()
+{
+   Serial.print("                                                              your score is ");
+   Serial.print(score);
+   Serial.print("\n \n");
 }
 
-void loop() { 
-  int b = randomgen();
-  switch (a) 
+void loop() 
+{ 
+  int b = randomGenLed();
+  int t = randomGenTime();
+  int timeleft = timer.limitValue - (timer.newTimeValue - timer.oldTimeValue);
+  switch (gameState) 
   {
-  case 1: // the game start
+  case GAME_SET_N_START: // the game start
         Serial.print("                                                        Please press Start Button!!!\n\n");
-        buttsWait();
-        timer.setTimer(15000); //сколько будет идти игра
-        timer.updateTimer(); // включаем таймер
-        a = 2;
+        startButtonWait(); //waiting for start button

```

Comment:
what will happen - if the player will not press the START_BUTTON here ?
will the system react on this ?

Reply:
system will wait for a start_button press for a decades